go/api/options/bitcount_options.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package options
+
+import (
+	"fmt"
+
+	"github.com/valkey-io/valkey-glide/go/glide/utils"
+)
+
+type BitmapIndexType string
+
+const (
+	BYTE BitmapIndexType = "BYTE"
+	BIT  BitmapIndexType = "BIT"
+)
+
+// Optional arguments to `BitCount` in [BitMapCommands]
+type BitCountOptions struct {
+	Start           *int64
+	End             *int64
+	BitMapIndexType BitmapIndexType
+}
+
+// SetStart defines start byte to calculate bitcount in bitcount command.
+func (options *BitCountOptions) SetStart(start int64) *BitCountOptions {
+	options.Start = &start
+	return options
+}
+
+// SetEnd defines start byte to calculate bitcount in bitcount command.
+func (options *BitCountOptions) SetEnd(end int64) (*BitCountOptions, error) {
+	if options.Start == nil {
+		return options, fmt.Errorf("End value cannot be set without start value, use SetStart()")
+	}
+	options.End = &end
+	return options, nil
+}
+
+// SetBitmapIndexType to specify start and end are in BYTE or BIT
+func (options *BitCountOptions) SetBitmapIndexType(bitMapIndexType BitmapIndexType) (*BitCountOptions, error) {
+	if options.Start == nil || options.End == nil {
+		return options, fmt.Errorf(
+			"SetBitmapIndexType value cannot be set without start and end values, use SetStart() and SetEnd()",
```

Comment:
As an option you can create factory methods for `BitCountOptions`: `BitCountOptionsWithStart`, `BitCountOptionsWithStartAndEnd`, `BitCountOptionsWithStartEndAndType` or so. This will make API more aligned with others.
You can use builder pattern as an alternative.

Reply:
But I don't insist on changes.